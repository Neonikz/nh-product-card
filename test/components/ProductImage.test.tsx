import React from 'react';
import renderer from 'react-test-renderer'
import { ProductCard } from '../../src/components/ProductCard';
import { ProductImage } from '../../src/components/ProductImage';
import { ProductTitle } from '../../src/components/ProductTitle';
import { product2 } from '../data/products';



describe('ProductImage Test', () => { 

    test('should show the component with its custom title', () => { 

        const wrapper = renderer.create(
            <ProductImage img="https://hola.jpg" />
        )
            expect(wrapper.toJSON()).toMatchSnapshot()
        })
        
        test('should show the component with its product name', () => {
            const wrapper = renderer.create(
                <ProductCard product={product2}>
                    {
                        () => (
                            <ProductTitle />
                        )
                    }
                </ProductCard>
            )
                expect(wrapper.toJSON()).toMatchSnapshot()
            
     })

 })
