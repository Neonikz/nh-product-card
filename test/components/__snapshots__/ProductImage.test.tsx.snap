// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Test should show the component with its custom title 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;

exports[`ProductImage Test should show the component with its product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Product Card con imagen
  </span>
</div>
`;
